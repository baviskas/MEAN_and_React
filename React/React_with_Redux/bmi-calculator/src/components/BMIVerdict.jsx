import React from 'react';

const BMIVerdict = ({BMI}) => {
    if(Number.isNaN(BMI)) {
        return null;
    } else {
        const bmiIndex = parseFloat(BMI);
        let verdict = 'your weight is normal.';
        if (bmiIndex < 18.5) {
            verdict = 'you are underweight.';
        } else if (bmiIndex>=25 && bmiIndex<=29.9) {
            verdict = 'you are overweight.';
        } else if (bmiIndex >= 30) {
            verdict = 'you are Obese.';
        }
        return <p>Your BMI is {BMI} and, {verdict}</p>;
    }
};

export default BMIVerdict;
