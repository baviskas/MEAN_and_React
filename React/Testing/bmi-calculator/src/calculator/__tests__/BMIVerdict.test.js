import { shallowWrapper } from "./../../testUtil";
import BMIVerdict from "../BMIVerdict";

const requiredProps = {
    bmi: 24,
};
const wrapComponent = (props) => {
    return shallowWrapper(BMIVerdict, requiredProps, props);
};

describe("VinInput: ", () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    describe("Smoke Test", () => {
        it("render smoketest for normal weight", () => {
            const wrapper = wrapComponent();
            expect(wrapper.debug()).toMatchSnapshot();
            console.log(wrapper.debug());
        });
        it("render smoketest for over weight", () => {
            const wrapper = wrapComponent({ bmi: 29 });
            expect(wrapper.debug()).toMatchSnapshot();
        });
        it("render smoketest for under weight", () => {
            const wrapper = wrapComponent({ bmi: 17 });
            expect(wrapper.debug()).toMatchSnapshot();
        });
        it("render smoketest for Obese", () => {
            const wrapper = wrapComponent({ bmi: 32 });
            expect(wrapper.debug()).toMatchSnapshot();
        });
        it("render smoketest for NAN", () => {
            const wrapper = wrapComponent({ bmi: NaN });
            expect(wrapper.debug()).toMatchSnapshot();
        });
    });
});
