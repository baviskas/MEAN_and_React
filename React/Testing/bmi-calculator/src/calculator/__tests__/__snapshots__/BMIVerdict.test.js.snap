// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VinInput:  Smoke Test render smoketest for NAN 1`] = `""`;

exports[`VinInput:  Smoke Test render smoketest for Obese 1`] = `
"<p>
   BMI is 
  32
  . 
  You are Obese
   
</p>"
`;

exports[`VinInput:  Smoke Test render smoketest for normal weight 1`] = `
"<p>
   BMI is 
  24
  . 
  You are Normal
   
</p>"
`;

exports[`VinInput:  Smoke Test render smoketest for over weight 1`] = `
"<p>
   BMI is 
  29
  . 
  You are Overweight
   
</p>"
`;

exports[`VinInput:  Smoke Test render smoketest for under weight 1`] = `
"<p>
   BMI is 
  17
  . 
  You are UnderWeight
   
</p>"
`;
